hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/BlockManagerImpl.java

```diff
@@ -101,8 +103,11 @@ public BlockManagerImpl(final ConfigurationSource conf,
     mxBean = MBeans.register("BlockManager", "BlockManagerImpl", this);
 
     // SCM block deleting transaction log and deleting service.
-    deletedBlockLog = new DeletedBlockLogImpl(conf, scm.getContainerManager(),
-        scm.getScmMetadataStore());
+    deletedBlockLog = new DeletedBlockLogImplV2(conf, scm.getContainerManager(),
+        scm.getScmHAManager().getRatisServer(),
+        scm.getScmMetadataStore().getDeletedBlocksTXTable(),
+        scm.getScmHAManager().getDBTransactionBuffer());
+    scmContext.setDeletedBlockLog(deletedBlockLog);
```

Comment:
This change ` scmContext.setDeletedBlockLog(deletedBlockLog);` is not needed.

You can use scm as a mediator.
```
StorageContainerManager.getScmBlockManager()
BlockManagerImpl.getDeletedBlockLog()
```